src/api/v3/role/mod.rs

```diff
@@ -13,23 +13,25 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use axum::{
-    extract::{Path, Query, State},
+    extract::{Path, Query, State,Json},
     response::IntoResponse,
 };
 use utoipa_axum::{router::OpenApiRouter, routes};
+use validator::Validate;
 
 use crate::api::auth::Auth;
 use crate::api::error::KeystoneApiError;
 use crate::assignment::AssignmentApi;
 use crate::keystone::ServiceState;
-use types::{Role, RoleList, RoleListParameters, RoleResponse};
+use types::{Role, RoleList, RoleListParameters, RoleResponse,RoleCreate};
 
 pub mod types;
 
 pub(crate) fn openapi_router() -> OpenApiRouter<ServiceState> {
     OpenApiRouter::new()
         .routes(routes!(list))
         .routes(routes!(show))
+        .routes(routes!(create))
```

Comment:
`create` and `list` belong together since they are technically the same URL just with different methods. Look in other apis